File: framework/languages/dotnet/samples/SupportChat/Server/Support/Adapters/ZLink/Spots/Handlers/SendChatMessageHandler.cs
using SupportChat.Server.Support.Adapters.ZLink.Actors;
using SupportChat.Shared.Contracts;
using Zlink.Framework.Contracts.Spots;

namespace SupportChat.Server.Support.Adapters.ZLink.Spots.Handlers;

internal sealed class SendChatMessageHandler
    : IZLinkSpotActorRequestHandler<ConversationSpot, SupportUserActor, SendChatMessageReq, SendChatMessageRes>
{
    public async ValueTask<SendChatMessageRes> HandleAsync(
        ConversationSpot spot,
        SupportUserActor actor,
        ZLinkSpotActorRequestContext context,
        SendChatMessageReq message,
        CancellationToken cancellationToken)
    {
        _ = context;
        return await spot.SendMessageAsync(actor, message, cancellationToken);
    }
}

File: framework/languages/java/samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/adapters/zlink/spots/handlers/SendChatMessageHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.adapters.zlink.spots.handlers

import systems.zlink.framework.CancellationToken
import systems.zlink.framework.kotlin.ZLinkSuspendingSpotActorRequestHandler
import systems.zlink.framework.spots.ZLinkSpotActorRequestContext
import systems.zlink.samples.kotlin.supportchat.server.support.adapters.zlink.actors.SupportUserActor
import systems.zlink.samples.kotlin.supportchat.server.support.adapters.zlink.spots.ConversationSpot
import systems.zlink.samples.kotlin.supportchat.shared.contracts.SendChatMessageReq
import systems.zlink.samples.kotlin.supportchat.shared.contracts.SendChatMessageRes

class SendChatMessageHandler() : ZLinkSuspendingSpotActorRequestHandler<
    ConversationSpot,
    SupportUserActor,
    SendChatMessageReq,
    SendChatMessageRes,
    > {
    override suspend fun handle(
        spot: ConversationSpot,
        actor: SupportUserActor,
        context: ZLinkSpotActorRequestContext,
        request: SendChatMessageReq,
        cancellationToken: CancellationToken,
    ): SendChatMessageRes =
        spot.sendMessage(actor, request)
}
